sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java

```diff
@@ -55,6 +59,29 @@ public String getBioentityInformation(@PathVariable String geneId) {
                         geneName,
                         propertyValues);
 
-        return model.get("bioentityProperties").toString();
+
+        // Removed last bracket '[' from the return Json to append the AtlasUrl at the end of this Json
+        String bioentityProperties = model.get("bioentityProperties").toString();
```

Comment:
Patching serialized representations of objects is dangerous and error prone. I strongly discourage you to do this. Start with `JsonArray jsonArray = GSON.fromJson(model.get("bioentityProperties").toString(), JsonArray.class);` and work with `jsonArray`.

Reply:
Agreed. Will update the code according to suggestion given.